src/logic/map_objects/tribes/worker_program.cc

```diff
@@ -343,17 +348,27 @@ void WorkerProgram::parse_findobject(Worker::Action* act, const std::vector<std:
 			   NoteMapObjectDescription(item.second, NoteMapObjectDescription::LoadType::kAttribute));
 			act->iparam2 = MapObjectDescr::get_attribute_id(item.second);
 		} else if (item.first == "type") {
-			act->sparam1 = item.second;
+			if (item.second == "immovable" || item.second == "bob" || item.second == "special") {
+				act->sparam1 = item.second;
+			} else {
+				throw GameDataError("Invalid usage of 'type' predicate: Possible values are 'immovable', 'bob' and 'special'.");
```

Comment:
I think the closing `}` is missing here for the `else` in line 353, that's why the next formatting run messed up the whole file. You're probably better of reverting this file to this commit (0fa64e2d76f7bb6de68e5f64f7b736234d483e78) for fixing it, otherwise it would be very hard to undo all the damage in the doc.

Reply:
OMG. what a stupid mistake. thanks for the advice